lib/file_store/middleware/logger.ex

```diff
@@ -45,9 +45,9 @@ defmodule FileStore.Middleware.Logger do
       |> log("RENAME", src: src, dest: dest)
     end
 
-    def upload(store, source, key) do
+    def upload(store, source, key, opts) do
```

Comment:
If we add this as a required positional argument, wouldn't that be a breaking change? Should we preserve `upload/3`?

Reply:
It won't be. The default `opt \\ []` will fill that in.